Improve MI performance doc (#3813)
wso2/docs-apim: en/docs/install-and-setup/setup/deployment-best-practices/performance-test-results-mi.md

- Throughput: The number of requests that the WSO2 Micro Integrator processes during a specific time interval (e.g. per second).. 
- Response Time: The end-to-end latency for an operation of invoking a service in WSO2 Micro Integrator. The complete distribution of response times was recorded.

!!! Note
    
    All the performance results were done on the default Micro Integrator distribution. The
    following performance results may become invalid if configurations are changed. For
    example, the default passthru io_buffer_size is 16KB. Hence the throughput significantly
    decreases for payload sizes above 16KB. If the buffer size is increased, the throughput

>     might differ. Similarly when worker pool thread counts are changed, the performance of

Reviewer: ```suggestion
    might differ. Similarly, when worker pool thread counts are changed, the performance of
```

>     Micro Integrator changes significantly. Hence, for accurate results doing a performance

Reviewer: ```suggestion
    Micro Integrator changes significantly. Therefore, to get accurate results, it is necessary to specify the exact configurations and scenarios used for the performance tests.
```

>     test with the exact configuration and scenarios is a must.

Reviewer: ```suggestion
```

## Deployment used for the test

> [![MI performance test all-in-one deployment]({{base_path}}/assets/img/setup-and-install/performance-test-results/mi_performance_test_all_in_one_deployment.png)]({{base_path}}/assets/img/setup-and-install/performance-test-results/mi_performance_test_all_in_one_deployment.png)

Reviewer: ```suggestion
Given below are the details of the deployment that was used for the performance test this is recorded in this documentation.

[![MI performance test all-in-one deployment]({{base_path}}/assets/img/setup-and-install/performance-test-results/mi_performance_test_all_in_one_deployment.png)]({{base_path}}/assets/img/setup-and-install/performance-test-results/mi_performance_test_all_in_one_deployment.png)
```

Author: `this is recorded in this documentation` or `that is recorded in this documentation`

Reviewer: good catch..

- [https://github.com/wso2/performance-common](https://github.com/wso2/performance-common)

- [https://github.com/wso2/performance-ei](https://github.com/wso2/performance-ei)

## Results

> 

Reviewer: ```suggestion
Given below is a summary of the performance result for each of the scenarios used for the tests.
```

The complete results can be found [here](https://github.com/wso2/micro-integrator/blob/b06581ed31fceaa32c01a03a63a107141a68cb2b/performance/benchmarks/summary.md).

#### Throughput (requests/sec) vs. concurrent users

> 

Reviewer: ```suggestion
The following results compare the throughput against the number of concurrent users in each scenario.
```

##### Direct Proxy

<img src="{{base_path}}/assets/img/setup-and-install/performance-test-results/mi/direct-proxy-tp.png)]({{base_path}}/assets/img/setup-and-install/performance-test-results/mi/direct-proxy-tp.png" alt="Direct Proxy Throughput" title="Direct Proxy Throughput" width="70%" />

##### Direct API

<img src="{{base_path}}/assets/img/setup-and-install/performance-test-results/mi/direct-api-tp.png)]({{base_path}}/assets/img/setup-and-install/performance-test-results/mi/direct-api-tp.png" alt="Direct API Throughput" title="Direct API Throughput" width="70%" />

##### CBR Transport Header Proxy

<img src="{{base_path}}/assets/img/setup-and-install/performance-test-results/mi/cbr-transport-header-proxy-tp.png" alt="CBR Transport Header Proxy Throughput" title="CBR Transport Header Proxy Throughput" width="70%" />

##### XSLT Proxy

<img src="{{base_path}}/assets/img/setup-and-install/performance-test-results/mi/xslt-proxy-tp.png" alt="XSLT Proxy Throughput" title="XSLT Proxy Throughput" width="70%" />

#### Average response time (ms) vs. concurrent users

> 

Reviewer: ```suggestion
The following results compare the average response time of the Micro Integrator server against the number of users in each scenario.
```
